Automattic/wp-calypso
# Domains: Add domain diagnostics card

## Proposed Changes

This PR adds the domain diagnostics card to the domain management page. That card will initially contain information about issues with misconfigured email DNS records (SPF, DKIM and DMARC). In the future, we'll add more information to it.

Depends on D134255-code.

### Screenshots

- Domain that doesn't use WPCOM name servers

> <img width="700" alt="Screenshot 2024-01-15 at 17 49 49" src="https://github.com/Automattic/wp-calypso/assets/5324818/7cab097f-9779-4385-af90-1f25fdc79a60">

- Domain that uses WPCOM name servers

> <img width="704" alt="Screenshot 2024-01-15 at 17 48 48" src="https://github.com/Automattic/wp-calypso/assets/5324818/df692afd-47b2-4a88-8e24-4ceb7bd1aed2">

- Confirmation dialog when you click the "Dismiss" button

> <img width="901" alt="Screenshot 2024-01-15 at 17 51 38" src="https://github.com/Automattic/wp-calypso/assets/5324818/6ad86f7d-992b-4a77-ba78-6c1e35b5c24f">

- If you dismiss the notice, the diagnostics card will be closed by default when entering the domain management page

> <img width="813" alt="Screenshot 2024-01-15 at 19 08 36" src="https://github.com/Automattic/wp-calypso/assets/5324818/7e5a34da-2aaa-44ea-8a80-805bac5b6b7d">

## Testing Instructions

- Apply D134255-code to your backend
- Open the live Calypso link or build this branch locally
- Open the domain management page for the primary domain in an atomic site or hack your backend so that the diagnostics endpoint thinks your site is atomic
- Change your DNS email records (SPF, DKIM 1 and 2, DMARC) to incorrect values (see pcYYhz-1KI-p2 for the expected values)
- Ensure the diagnostics card shows the correct diagnostic for each value
- In the backend, try setting the `is_using_wpcom_name_servers` attribute to true and false to ensure the instructions to fix the issues are correct in each case
- In the case the domain uses WPCOM name servers, click on the "Fix DNS issues automatically" button and ensure the correct DNS records are added to your domain
- Ensure the diagnostics card is not shown anymore after the records are fixed

Also please test the "Dismiss" action in the notice
- When the notice is dismissed, the diagnostics card should be closed by default when loading the domain management page
- Use this code (332c5-pb) to reset the dismissed notice state from the backend

## Pre-merge Checklist
